### PR TITLE
Pure Python variant with test setup

### PR DESCRIPTION
**Pull request by mlange05**
_Note: This is the first in a list of planned PRs to homogenise the various Python variants we're investigating, so it won't be complete in the full sense, but instead pave the way for further infrastructure work._

The key contribution of this PR is the setup of a completely standalone pure Python variant of CLOUDSC that uses vanilla numpy array notation and produces correct results against the reference file. The kernel was originally generated via Loki, and there is a hidden option to re-generate the kernel file on-the-fly. The key idea here is that this will pave the way for a "pure on-the-fly JIT" variant that could generate Python-wrapped fortran kernels (without CMake setup), to slot into the provided Python driver. 

The initial setup, code-named `cloudscf2py` hsa also been set up for automatic CI setup and intergates with the bundle/cmake setup we use for the other variants. Unlike `cloudscpyiface` this uses CMake's `add_custom_target` and a full dependency chain of custom commands to install the alternative driver and symlink it into the build.

Further homogenisation with the PyIface variant is planned, but left for a follow-on PR.

The PR also puts the new driver/package into the same pip install as the `cloudsc4py` variant (and switches to `pyproject.toml` pip setup), but the GT4Py-based variant is not exposed to the CI runner, as it will likely be temporarily removed soon; again in a follow-on PR.  

This PR also removes the previously introduced pre-commit hooks, as they have become a little tedious on the Fortran development side.

### CODE REVIEW

File: src/cloudsc_python/drivers/cloudsc_f2py.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    from cloudscf2py import (
+        load_input_fields, load_input_parameters, load_reference_fields,
+        cloudsc_py
+    )
+
+    fields = load_input_fields(path=input_path)
+
+    yrecldp, yrmcst, yrethf, yrephli, yrecld = load_input_parameters(path=input_path)
+
+    cloudsc_args = {
```
Comment (reuterbal) [inline src/cloudsc_python/drivers/cloudsc_f2py.py:89]: This looks a little odd - should there be default values in `cloudsc_args` or could this be merged with the statement below?

File: src/cloudsc_python/drivers/cloudsc_f2py.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+            ' {absavg:20.13e}  {maxrel:20.13e}'.format(
+                fname=name.upper(), fmin=f.min(), fmax=f.max(),
+                absmax=np.absolute(f - ref).max(),
+                absavg=np.sum(np.absolute(f - ref)) / ngptot,
+                maxrel=0.0 if zerrsum < zeps else (zerrsum/(1.0+zsum) if zsum < zeps else zerrsum/zsum)
+            )
+        )
+
+
+def run_cloudsc_kernel(nthreads, ngptot, nproma, input_path, reference_path):
```
Comment (reuterbal) [inline src/cloudsc_python/drivers/cloudsc_f2py.py:79]: Am I missing something here or is this missing the input field expansion? Which means, any value for `ngptot` > 100 will not work?
Also, `nthreads` is not used (yet?).

**Review by reuterbal — CHANGES_REQUESTED** [review]
This looks great and the CMake integration is pretty neat!

I've flagged two or three small issues which would be good to resolve. And I've left a few questions/remarks that touch more on my understanding what the strategy is here. Let me know what your thoughts are there.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -17,7 +17,7 @@ on:
 jobs:
   # This workflow contains a single job called "build"
   build:
-    name: Test on ${{ matrix.arch }} ${{ matrix.io_library_flag }} ${{ matrix.mpi_flag }} ${{ matrix.prec_flag }} ${{ matrix.gpu_flag }} ${{ matrix.loki_flag }} ${{ matrix.claw_flag }} ${{ matrix.pyiface_flag }}
+    name: Test on ${{ matrix.arch }} ${{ matrix.io_library_flag }} ${{ matrix.mpi_flag }} ${{ matrix.prec_flag }} ${{ matrix.gpu_flag }} ${{ matrix.loki_flag }} ${{ matrix.claw_flag }} ${{ matrix.pyiface_flag }} ${{ matrix.python_flag }}
```
[the remark this answers is not in the document]
Comment (mlange05) [reply .github/workflows/build.yml:20]: Yes, was planning to homogenise all the python variants under a single flag, but that will come later. And you're right, for now we should make it explicit.

File: src/cloudsc_python/src/cloudscf2py/cloudsc.F90
[the remark this answers is not in the document]
Comment (mlange05) [reply src/cloudsc_python/src/cloudscf2py/cloudsc.F90:1]: No, there are actually subtle restrictions for Python code that do not apply to the other variants - mainly due to the fact that we cannot access support modules and need to pass everything through the argument list at all times. I agree about the degree of duplication, but it's often warranted and meant to show the restrictions for individual cases.

**Review by reuterbal — COMMENTED** [review]
Thanks for addressing the comments! I think you have accidentally disabled the testing of this variant now because `python_flag` has not actually been renamed in the build matrix but only in some of the reporting (which is simply expanded to an empty string).

**Pull request merged**